### PR TITLE
Fix upload big xml files for tasks

### LINKED ISSUE #123: Uploading large XML Files
I am having issues uploading large XML files. I created a task with 130K images. I uploaded the annotations XML file around 120MB. It says file being processed. Once it says annotations are being saved, it fails. Changed the DATA_UPLOAD_MAX_MEMORY_SIZE value to 300 MB from the default 100MB but still no luck.

### PR DESCRIPTION
**Pull request by nmanovic**
Fix #123.
- I don't like the solution. It isn't good split annotations on several parts and submit one by one. Need to find a better solution.
- Add good profiler: Silk
- Optimize a little bit save_job

### CODE REVIEW

**Comment by nmanovic** [comment]
@azhavoro, @bsekachev,

Please review. It seems to split on chunks isn't a bad idea. I implemented the approach only when we upload annotations for a task. In general need to send "chunks" when we upload data for a job and when we save annotations for a job. In the latest case need to be sure that we update internal state correctly.

File: cvat/apps/dashboard/static/dashboard/js/dashboard.js [hunk 1]
```diff
@@ -538,29 +538,62 @@ function uploadAnnotationRequest() {
...
                     }
 
-                    const exportData = createExportContainer();
-                    exportData.create = parsed;
-
                     let asyncSave = function() {
                         $.ajax({
-                            url: '/save/annotation/task/' + window.cvat.dashboard.taskID,
+                            url: '/delete/annotation/task/' + window.cvat.dashboard.taskID,
                             type: 'POST',
```
Comment (bsekachev) [inline cvat/apps/dashboard/static/dashboard/js/dashboard.js]: I think method DELETE is better for delete request

File: cvat/apps/dashboard/static/dashboard/js/dashboard.js [hunk 2]
```diff
@@ -538,29 +538,62 @@ function uploadAnnotationRequest() {
...
                         overlay.remove();
-                        showMessage("Parsing errors was occured. " + error);
+                        showMessage("Parsing errors was occurred. " + error);
                         return;
                     }
 
-                    const exportData = createExportContainer();
-                    exportData.create = parsed;
-
                     let asyncSave = function() {
```
Comment (bsekachev) [inline cvat/apps/dashboard/static/dashboard/js/dashboard.js]: This name became unsuitable

File: cvat/apps/dashboard/static/dashboard/js/dashboard.js [hunk 3]
```diff
@@ -538,29 +538,62 @@ function uploadAnnotationRequest() {
...
-                            data: JSON.stringify(exportData),
-                            contentType: 'application/json',
                             success: function() {
-                                let message = 'Annotation successfully uploaded';
-                                showMessage(message);
+                                asyncSaveChunk(0);
                             },
                             error: function(response) {
-                                let message = 'Annotation uploading errors was occured. ' + response.responseText;
+                                let message = 'Annotation uploading errors was occurred: ' +
```
Comment (bsekachev) [inline cvat/apps/dashboard/static/dashboard/js/dashboard.js]: There is no data uploading. Clean up only.

File: cvat/apps/dashboard/static/dashboard/js/dashboard.js [hunk 4]
```diff
@@ -538,29 +538,62 @@ function uploadAnnotationRequest() {
...
+                                overlay.remove();
                             },
-                            complete: () => overlay.remove()
                         });
+                    }
+
+                    let asyncSaveChunk = function(start) {
+                        let CHUNK_SIZE = 100000;
+                        let end = start + CHUNK_SIZE;
+                        var chunk = {};
```
Comment (bsekachev) [inline cvat/apps/dashboard/static/dashboard/js/dashboard.js]: let's use only "let" variables. It's more safe

File: cvat/apps/dashboard/static/dashboard/js/dashboard.js [hunk 5]
```diff
@@ -538,29 +538,62 @@ function uploadAnnotationRequest() {
...
+                                    response.responseText;
                                 showMessage(message);
+                                overlay.remove();
                             },
-                            complete: () => overlay.remove()
                         });
+                    }
+
+                    let asyncSaveChunk = function(start) {
+                        let CHUNK_SIZE = 100000;
```
Comment (bsekachev) [inline cvat/apps/dashboard/static/dashboard/js/dashboard.js]: there is "const" specificator

File: cvat/apps/dashboard/static/dashboard/js/dashboard.js [hunk 6]
```diff
@@ -538,29 +538,62 @@ function uploadAnnotationRequest() {
...
+                                url: '/save/annotation/task/' + window.cvat.dashboard.taskID,
+                                type: 'POST',
+                                data: JSON.stringify(exportData),
+                                contentType: 'application/json',
+                                success: function() {
+                                    asyncSaveChunk(end);
+                                },
+                                error: function(response) {
+                                    let message = 'Annotation uploading errors were occurred: ' +
+                                        response.responseText;
```
Comment (bsekachev) [inline cvat/apps/dashboard/static/dashboard/js/dashboard.js]: What about "/delete/annotation/task/" request in this case?

File: cvat/apps/engine/annotation.py [hunk 7]
```diff
@@ -117,7 +143,35 @@ def save_task(tid, data):
...
+        # if an item inside _data isn't empty need to call save_job
+        isNonEmpty = False
+        for action in ['create', 'update', 'delete']:
+            for objects in _data[action].values():
+                if objects:
+                    isNonEmpty = True
+                    break
+
+        if isNonEmpty:
+            save_job(jid, _data, True)
```
Comment (bsekachev) [inline cvat/apps/engine/annotation.py]: What will be happened if data from single job are separated to two chunks?
Third "True" argument might remove data from previous request.

Comment (bsekachev) [reply cvat/apps/engine/annotation.py]: Why do we need erase old data if we called "delete" request from client before?
isNonEmpty flag doesn't appear pretty

File: cvat/apps/dashboard/static/dashboard/js/dashboard.js (hunk 2 above)
Comment (nmanovic) [reply cvat/apps/dashboard/static/dashboard/js/dashboard.js]: asyncSave is a good name here. Let's keep it.

File: cvat/apps/dashboard/static/dashboard/js/dashboard.js (hunk 1 above)
Comment (nmanovic) [reply cvat/apps/dashboard/static/dashboard/js/dashboard.js]: OK

File: cvat/apps/dashboard/static/dashboard/js/dashboard.js (hunk 3 above)
Comment (nmanovic) [reply cvat/apps/dashboard/static/dashboard/js/dashboard.js]: OK

File: cvat/apps/dashboard/static/dashboard/js/dashboard.js (hunk 4 above)
Comment (nmanovic) [reply cvat/apps/dashboard/static/dashboard/js/dashboard.js]: OK

**Pull request merged**